test/services/lessons/index.test.js

```diff
@@ -107,10 +163,107 @@ describe('lessons service', () => {
+		const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+		const course = await testObjects.createTestCourse({ schoolId, teacherIds: [teacher._id] });
+		const lesson = await testObjects.createTestLesson({ name: 'testlesson', courseId: course._id });
+		const params = await testObjects.generateRequestParamsFromUser(teacher);
+		const data = {
+			name: 'this name is better',
+		};
+		const result = await app.service('lessons').patch(lesson._id, data, params);
+		expect(result._id.toString()).to.equal(lesson._id.toString());
+		expect(result.name).to.equal('this name is better');
+	});
+
+	it('Teacher can not PATCH lesson on foreign school', async () => {
+		const { _id: schoolId } = await testObjects.createTestSchool({});
+		const { _id: otherschoolId } = await testObjects.createTestSchool({});
+		const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+		const otherTeacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId: otherschoolId });
+		const course = await testObjects.createTestCourse({ otherschoolId, teacherIds: [otherTeacher._id] });
+		const lesson = await testObjects.createTestLesson({ name: 'testlesson', courseId: course._id });
+		const params = await testObjects.generateRequestParamsFromUser(teacher);
+		const data = {
+			name: 'hostile takeover',
+		};
+		try {
+			await app.service('lessons').patch(lesson._id, data, params);
+			throw new Error('should have failed');
+		} catch (err) {
+			expect(err.message).to.not.equal('should have failed');
+			expect(err.code).to.equal(404);
+			expect(err.message).to.equal(`no record found for id '${lesson._id}'`);
+		}
+	});
+
+	it('Teacher can not UPDATE lesson on foreign school', async () => {
+		const { _id: schoolId } = await testObjects.createTestSchool({});
+		const { _id: otherschoolId } = await testObjects.createTestSchool({});
+		const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+		const otherTeacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId: otherschoolId });
+		const course = await testObjects.createTestCourse({ otherschoolId, teacherIds: [otherTeacher._id] });
+		const targetCourse = await testObjects.createTestCourse({ schoolId, teacherIds: [teacher._id] });
+		const lesson = await testObjects.createTestLesson({ name: 'testlesson', courseId: course._id });
+		const params = await testObjects.generateRequestParamsFromUser(teacher);
+		const data = {
+			name: 'hostile takeover',
+			courseId: targetCourse._id,
+		};
+		try {
+			await app.service('lessons').update(lesson._id, data, params);
+			throw new Error('should have failed');
+		} catch (err) {
+			expect(err.message).to.not.equal('should have failed');
+			expect(err.code).to.equal(404);
+			expect(err.message).to.equal(`no record found for id '${lesson._id}'`);
+		}
+	});
+
+	it('Teacher can not REMOVE lesson on foreign school', async () => {
```

Comment:
Test that student can delete, patch, create CourseGroup lessons are very helpful in context of hotfix deployment.